GoogleForCreators/web-stories-wp
# Shopping: Prevent duplicate products on a page

## Context

It is makes for invalid story by having duplicate products on the same page. 

## Summary



## Relevant Technical Choices

Tweak the reducer to not allow for new element to be created / duplicate with the same id. 

## To-do



## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

1. Create a story. 
2. Add a product. 
3. Select product element. 
4. Copy element cmd + c
5. Deselect product
6. paste element cmd + v. 
7. Product element should **not** be duplicated. 

Right click should not be able to duplicate product.
Layer panel  should not be able to duplicate product.


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #11521
